tests/Owin.Security.OpenIdConnect.Server.Tests/OpenIdConnectAuthorizationRequiredParameterTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.Owin.Testing;
+using Xunit;
+
+namespace Owin.Security.OpenIdConnect.Server.Tests {
+    /// <summary>
+    ///     This class tests a bunch of wrong combination calling Authorization Endpoint,
+    ///     source of truth is http://openid.net/specs/openid-connect-core-1_0.html#AuthorizationEndpoint
+    /// </summary>
+    public class OpenIdConnectAuthorizationRequiredParameterTests {
+        public static TheoryData<HttpMethod> AuthorizationEndpointHttpMethods {
+            get { return new TheoryData<HttpMethod> {HttpMethod.Get, HttpMethod.Post}; }
+        }
+
+        [Theory]
+        [MemberData("AuthorizationEndpointHttpMethods")]
```

Comment:
`MemberData` is pretty cool when you have many cases, but I guess it would be easier to use `InlineData` in these tests.